### PR TITLE
Parts Interaction Regression

### PR DESCRIPTION
**Pull request by gtrentalancia**
Please merge to fix a bug (item drag and drop does not work anymore).

### CODE REVIEW

**Comment by drahnr** [comment]
Did you test this yourself?

File: src/model/item-data.c [hunk 1]
```diff
@@ -249,15 +249,25 @@ void item_data_set_pos (ItemData *item_data, Coords *pos)
...
 		return;
 
+	if (!delta->x && !delta->y)
+		return;
+
 	priv = item_data->priv;
 	cairo_matrix_translate (&(priv->translate), delta->x, delta->y);
+
+	handler_connected =
+	    g_signal_handler_is_connected (G_OBJECT (item_data), item_data->changed_handler_id);
```
Comment (drahnr) [inline src/model/item-data.c]: The check if anything is connected is unnecessary.

File: src/model/item-data.c [hunk 2]
```diff
@@ -249,15 +249,25 @@ void item_data_set_pos (ItemData *item_data, Coords *pos)
...
 	ItemDataPriv *priv;
+	gboolean handler_connected;
 
 	g_return_if_fail (item_data != NULL);
 	g_return_if_fail (IS_ITEM_DATA (item_data));
 
 	if (delta == NULL)
 		return;
 
+	if (!delta->x && !delta->y)
```
Comment (drahnr) [inline src/model/item-data.c]: these are floats and thus this will always trigger, use `fabs(x) < EPS`

File: src/model/item-data.c (hunk 1 above)
Comment (gtrentalancia) [reply src/model/item-data.c]: That's fine, I've removed the above mentioned check.

However, there are similar checks elsewhere in the code.

**Comment by gtrentalancia** [comment]
I have tested it and the drag and drop feature is fully functional.

File: src/model/item-data.c (hunk 2 above)
Comment (gtrentalancia) [reply src/model/item-data.c]: Without it, it would perform unnecessary and very expensive matrix multiplication.

**Comment by gtrentalancia** [comment]
Since I have tested it several times and it works for me, can you please double-check ?

The only explanation I can find is that you have a buggy dependency installed (cairo ??).

**Comment by gtrentalancia** [comment]
For drag and drop functionality I mean:

- selecting an item (such as a part or wire);
- while keeping the mouse button pressed, move the item on the sheet to the desired position;
- release the mouse button.

At the end of the above operation, the item should be moved to the new position and snapped to the grid (e.g. connected to existing nearby wires). The should be no item in the original position (as in a "cut and paste" operation).

**Comment by gtrentalancia** [comment]
In a similar fashion, it is also possible to perform a "drag and drop" operation on a group of items (parts and wires) by first selecting the group of items (they should become green-coloured) and then performing the above mentioned steps as in the case of a single item.

File: src/model/item-data.c (hunk 2 above)
Comment (drahnr) [reply src/model/item-data.c]: Please read again what I wrote, the check is necessary, but it will never trigger as given above.

Comment (gtrentalancia) [reply src/model/item-data.c]: I've just amended it !

**Comment by drahnr** [comment]
I retested, and it works. Rotation seems to work better as well. 👍

**Pull request merged**